commons-numbers-complex-arrays/src/test/java/org/apache/commons/numbers/complex/arrays/ComplexListTest.java

```diff
@@ -126,18 +126,18 @@ void testAddAndAddAll() {
 
     @Test
     void testSetAddAndAddAllNullPointerException() {
+        ComplexList copy = ComplexList.interleaved();
         ComplexList list1 = generateComplexList(3);
         Assertions.assertThrows(NullPointerException.class, () -> list1.add(null));
         Assertions.assertThrows(NullPointerException.class, () -> list1.add(0, null));
         Assertions.assertThrows(NullPointerException.class, () -> list1.set(1, null));
 
-        ComplexList copy = ComplexList.interleaved();
-        copy.addAll(list1);
         List<Complex> list2 = generateList(3);
         list2.set(1, null);
         Assertions.assertThrows(NullPointerException.class, () -> list1.addAll(list2));
-        Assertions.assertEquals(copy.size(), list1.size());
         Assertions.assertThrows(NullPointerException.class, () -> list1.addAll(0, list2));
+        copy.addAll(list1);
```

Comment:
copy.addAll(list1) should be **before** all the operations on list1. Otherwise you are not comparing before (the copy) and after (list1).